components/director/pkg/sanitize/directive.go

```diff
@@ -0,0 +1,45 @@
+package sanitize
+
+import (
+	"context"
+
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/kyma-incubator/compass/components/director/pkg/log"
+	"github.com/kyma-incubator/compass/components/director/pkg/scope"
+	"github.com/kyma-incubator/compass/components/director/pkg/str"
+	"github.com/pkg/errors"
+)
+
+//go:generate mockery --name=ScopesGetter --output=automock --outpkg=automock --case=underscore
+type ScopesGetter interface {
+	GetRequiredScopes(scopesDefinition string) ([]string, error)
+}
+
+type directive struct {
+	scopesGetter ScopesGetter
+}
+
+func NewDirective(getter ScopesGetter) *directive {
+	return &directive{
+		scopesGetter: getter,
+	}
+}
+
+func (d *directive) Sanitize(ctx context.Context, obj interface{}, next graphql.Resolver, scopesDefinition string) (interface{}, error) {
```

Comment:
just an idea to remove some duplication 

Currently hasScopes and sanitize directives are exactly the same apart from what they return on scope missmatch (error forbidden vs nil error). You can have a single directive struct that accepts scopesGetter and "scopesMissmatchErrorProvider" or something with better name.  Then you can then reuse the same struct for the directive in the code for both instantiations passing different arguments.